File: src/main/java/com/natia/secretmod/core/BlockRenderingHook.java
package com.natia.secretmod.core;

import net.minecraft.init.Blocks;
import net.minecraft.util.BlockPos;
import net.minecraftforge.client.event.DrawBlockHighlightEvent;
import net.minecraftforge.client.event.RenderBlockOverlayEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class BlockRenderingHook {

    public static float globalPartialTicks;
    public static BlockPos skullBlockPos;

    @SubscribeEvent
    public void partialTicks(DrawBlockHighlightEvent event) {
        globalPartialTicks = event.partialTicks;
    }

    @SubscribeEvent
    public void drawBlock(RenderBlockOverlayEvent event) {
        if (event.blockForOverlay.getBlock().equals(Blocks.skull)) {
            skullBlockPos = event.blockPos;
        }
    }

}

File: src/main/kotlin/natia/skytweaks/core/BlockRenderingHook.kt
package natia.skytweaks.core

import net.minecraft.init.Blocks
import net.minecraft.util.BlockPos
import net.minecraftforge.client.event.DrawBlockHighlightEvent
import net.minecraftforge.client.event.RenderBlockOverlayEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class BlockRenderingHook {

    @SubscribeEvent
    fun partialTicks(event: DrawBlockHighlightEvent) {
        globalPartialTicks = event.partialTicks
    }

    @SubscribeEvent
    fun drawBlock(event: RenderBlockOverlayEvent) {
        if (event.blockForOverlay.block == Blocks.skull) {
            skullBlockPos = event.blockPos
        }
    }

    companion object {

        var globalPartialTicks: Float = 0.toFloat()
        var skullBlockPos: BlockPos? = null
    }

}
